nuxeo-runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/services/config/ConfigurationServiceImpl.java

```diff
@@ -65,15 +111,53 @@ public boolean isBooleanPropertyFalse(String key) {
     @Override
     public void registerContribution(Object contribution, String extensionPoint, ComponentInstance contributor) {
         if (CONFIGURATION_EP.equals(extensionPoint)) {
-            registry.addContribution((ConfigurationPropertyDescriptor) contribution);
+            descriptors = null;
+            ConfigurationPropertyDescriptor configurationPropertyDescriptor = (ConfigurationPropertyDescriptor) contribution;
+            String key = configurationPropertyDescriptor.getName();
+            if (Framework.getProperties().containsKey(key)) {
+                String message = "Property '" + key + "' should now be contributed to extension "
+                        + "point 'org.nuxeo.runtime.ConfigurationService', using target 'configuration'";
+                DeprecationLogger.log(message, "7.4");
+                Framework.getRuntime().getMessageHandler().addWarning(message);
+            }
+            super.registerContribution(contribution, extensionPoint, contributor);
         }
     }
 
     @Override
     public void unregisterContribution(Object contribution, String extensionPoint, ComponentInstance contributor) {
         if (CONFIGURATION_EP.equals(extensionPoint)) {
-            registry.removeContribution((ConfigurationPropertyDescriptor) contribution);
+            descriptors = null;
+            super.unregisterContribution(contribution, extensionPoint, contributor);
         }
     }
 
+    @Override
+    protected String getName() {
+        return COMPONENT_NAME;
+    }
+
+    @Override
+    public Map<String, Serializable> getProperties(String namespace) {
+        if (StringUtils.isBlank(namespace)) {
+            return null;
+        }
+        if (namespace.charAt(namespace.length() - 1) == '.') {
+            throw new IllegalArgumentException("namespace cannot end with a dot");
+        }
+        return getDescriptors().values()
+                               .stream()
+                               .filter(desc -> desc.getName().length() > namespace.length()
+                                       && desc.getName().startsWith(namespace)
+                                       && desc.getName().charAt(namespace.length()) == '.')
```

Comment:
Please make a helper function `boolean startsWithNamespace(String string, String namespace)` to avoid repeats.